packages/superset-ui-legacy-plugin-chart-sunburst/src/transformProps.js

```diff
@@ -18,14 +18,24 @@
  */
 /* eslint-disable sort-keys */
 export default function transformProps(chartProps) {
-  const { width, height, formData, queryData } = chartProps;
+  const { width, height, formData, queryData, datasource } = chartProps;
   const { colorScheme, metric, secondaryMetric } = formData;
 
-  return {
+  const returnProps = {
     width,
     height,
     data: queryData.data,
     colorScheme,
     metrics: [metric, secondaryMetric],
   };
+
+  if (datasource && datasource.metrics) {
```

Comment:
```ts
if (datasource && datasource.metrics) {
  const metricWithFormat = datasource.metrics.find(({ metric_name, d3format }) => metric_name === formData.metric && d3format);
  if (metricWithFormat) {
    Object.assign(returnProps, { numberFormat: metricWithFormat.d3format });
  } 
}
```

Reply:
Works for me. Committed this with a slight tweak to destructuring to make eslint happier: `...find(({ metric_name: metricName, d3format }) => metricName ...`